python-package/xgboost/data.py

```diff
@@ -247,7 +248,26 @@ def _is_modin_df(data: DataType) -> bool:
     "boolean": "i",
 }
 
+pandas_pyarrow_mapper = {
+    "int8[pyarrow]": "i",
+    "int16[pyarrow]": "i",
+    "int32[pyarrow]": "i",
+    "int64[pyarrow]": "i",
+    "uint8[pyarrow]": "i",
+    "uint16[pyarrow]": "i",
+    "uint32[pyarrow]": "i",
+    "uint64[pyarrow]": "i",
+    "float16[pyarrow]": "float",
```

Comment:
I don't think float16 is supported by pyarrow so this isn't supported by pandas

Reply:
Thank you for the information, removed the type.